coalib/testing/LocalBearTestHelper.py

```diff
@@ -52,6 +54,22 @@ class LocalBearTestHelper(unittest.TestCase):
     If you miss some methods, get in contact with us, we'll be happy to help!
     """
 
+    def assertComparableObjectsEqual(self, observed_result, expected_result):
+        self.assertEqual(len(observed_result), len(expected_result),
+                         msg='The number of elements in the observed result '
+                             'and expected result differ.')
```

Comment:
> To determine if it is only new or only deleted items, you must look at each expected item and find it in the resulting array

This might lead to problems if they don't match (i.e. the user makes a mistake in putting wrong values of the attributes ?)

However, I have a found a workaround that will work when both the lists are sorted (i.e. the `expected_result` and `observed_result`) : (Note that I have skipped the lower part i.e. outermost `else`)

```
    def assertComparableObjectsEqual(self, observed_result, expected_result):
        if len(observed_result) != len(expected_result):
            if len(observed_result) > len(expected_result):
                msg = ('The number of elements in the observed result({}) is '
                       'greater than the number of elements in the expected '
                       'result({}). The lacking elements in the expected '
                       'results are:\n'.format(len(observed_result),
                                               len(expected_result)))
                for i in range(
                     len(observed_result) - 1, len(expected_result) - 1, -1):
                    msg += (str(observed_result[i]) + '\n')
                raise AssertionError(msg)
            else:
                msg = ('The number of elements in the observed result({}) is '
                       'lesser than the number of elements in the expected '
                       'result({}). The extra elements in the expected '
                       'results are:\n'.format(len(observed_result),
                                               len(expected_result)))
                for i in range(
                     len(expected_result) - 1, len(observed_result) - 1, -1):
                    msg += (str(expected_result[i]) + '\n')
                raise AssertionError(msg)
```
When the results are sorted, the extra elements can be found out and reported which we've done here. There is another case when the results are not sorted because `check_order` is `True`. Here, we expect the users to enter the results in the same order. We can use a similar approach here too.

CC @jayvdb